Adding docstring for IRHeader

maddyonline opened the pull request:
I used http://stackoverflow.com/a/20573384 as a guide to write docstring for namedtuple directly beneath it.

I am not sure if this documentation actually adds clarity or is in fact not required. Either way, creating a PR for any comments. (Feel free to close, if this is not required.)

@mli @zackchase @madjam @nswamy

piiswrong commented on python/mxnet/recordio.py:
```diff
@@ -166,6 +166,19 @@ def write_idx(self, idx, buf):
+id: int
+    usually a unique id representing record
+id2: int
+    available for convenience, can be set arbitrarily
```
I think this is the higher bits of id. Should be 0

maddyonline replied:
Fixed. Thanks. Let me know if a better comment is in order.

zackchase replied:
-> "Available for convenience, can be set arbitrarily."

zackchase commented on python/mxnet/recordio.py:
```diff
@@ -166,6 +166,19 @@ def write_idx(self, idx, buf):
 
 
 IRHeader = namedtuple('HEADER', ['flag', 'label', 'id', 'id2'])
+"""An alias for HEADER used for storing metadata (e.g. labels) that accompanies a record.
```
Right now the sentence runs on a bit. Please change to:

"An alias for HEADER. Used to store metadata (e.g. labels) accompanying a record."

zackchase commented on python/mxnet/recordio.py:
```diff
@@ -166,6 +166,19 @@ def write_idx(self, idx, buf):
+Parameters
+----------
+flag : int
+    available for convenience, can be set arbitrarily
```
Capitalize: "Available for convenience ..."

zackchase replied:
end in period.

zackchase commented on python/mxnet/recordio.py:
```diff
@@ -166,6 +166,19 @@ def write_idx(self, idx, buf):
+flag : int
+    available for convenience, can be set arbitrarily
+label : float or an array of float
+    typically used to store label(s) for a record
```
Capitalize "Typically"

zackchase replied:
End in period.

zackchase commented on python/mxnet/recordio.py:
```diff
@@ -166,6 +166,19 @@ def write_idx(self, idx, buf):
+label : float or an array of float
+    typically used to store label(s) for a record
+id: int
+    usually a unique id representing record
```
-> "Usually a unique id representing record."

nswamy commented:
@maddyonline, Thanks for your contribution. could you please make the suggested changes so we could merge this?

maddyonline commented:
@zackchase Thanks for suggestions. @nswamy Addressed comments, see https://github.com/dmlc/mxnet/pull/6114/commits/7decb10721d02a3aa24ff520eefaf1baa7331187

Let me know if you want me to squash my commits before merging.

piiswrong commented on python/mxnet/recordio.py:
```diff
@@ -272,6 +272,19 @@ def write_idx(self, idx, buf):
+Parameters
+----------
+flag : int
+    Available for convenience, can be set arbitrarily.
```
I don't think this is true.

maddyonline replied:
What would be the correct documentation then?

piiswrong replied:
Actually I doesn't hurt for current implementation.
